From 2921bf50ea3e0eb3dfaac0895613feb14e52600c Mon Sep 17 00:00:00 2001
From: Wayne Zou <b36644@freescale.com>
Date: Mon, 8 Aug 2011 17:14:37 +0800
Subject: [PATCH 326/678] ENGR00154468 mx53 pcba: Add DC-IN power supply
 support for revB board

Add DC-IN power supply support for revB board when booting from EMMC.
set  both AUX&USB current limit to 1.5A for Ripley 2.1 only
Change CC current to 950mA

Signed-off-by: Wayne Zou <b36644@freescale.com>
---
 board/freescale/mx53_pcba/mx53_pcba.c |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/board/freescale/mx53_pcba/mx53_pcba.c b/board/freescale/mx53_pcba/mx53_pcba.c
index 3773ee4..011460a 100644
--- a/board/freescale/mx53_pcba/mx53_pcba.c
+++ b/board/freescale/mx53_pcba/mx53_pcba.c
@@ -330,11 +330,19 @@ void setup_pmic_voltages(void)
 			printf("%s:i2c_write:error\n", __func__);
 		}
 
+		/* set both AUX&USB current limit to 1.5A for Ripley 2.1 only */
+		if (i2c_read(0x8, 52, 1, &buf[0], 3))
+			printf("%s:i2c_read 52:error\n", __func__);
+		buf[0] = (buf[1] & 0xef) | 0x1;
+		if (i2c_write(0x8, 52, 1, buf, 3))
+			printf("%s:i2c_write 52:error\n", __func__);
+
+		/* Change CC current to 950mA */
 		/* Change CV voltage as 4.2v */
 		if (i2c_read(0x8, 51, 1, &buf[0], 3))
 			printf("%s:i2c_read 51:error\n", __func__);
 
-		buf[1] = (buf[1] & 0xf0) | 0x8;
+		buf[1] = (buf[1] & 0x0) | 0x78;
 		buf[2] = (buf[2] & 0x3f) | 0xc0;
 		if (i2c_write(0x8, 51, 1, buf, 3))
 			printf("%s:i2c_write 51:error\n", __func__);
-- 
1.7.7.4

